src/odinapi/utils/datamodel.py

```diff
@@ -0,0 +1,432 @@
+import attr
+from typing import List, Any, Dict
+import datetime as dt
+from enum import Enum
+from dateutil.relativedelta import relativedelta
+
+
+DATEFMT = "%Y-%m-%dT%H:%M:%SZ"
+
+
+COMMON_FILE_HEADER_DATA = {
+    "creator_name": 'Donal Murtagh',
+    "creator_url": 'odin.rss.chalmers.se',
+    "creator_email": 'donal.murtagh@chalmers.se',
+    "address": '412 96 Gothenburg, Sweden',
+    "institution": 'Chalmers University of Technology',
+    "platform": 'Odin',
+    "sensor": 'SMR',
+    "version_l1b": "8",
+    "version_l2": "3.0.0"
+}
+
+
+class L2Type(Enum):
+    l2 = "l2"
+    l2i = "l2i"
+    l2anc = "l2anc"
+
+
+class DType(Enum):
+    i8 = "i8"
+    f4 = "f4"
+    double = "double"
+
+
+class Dimension(Enum):
+    d1 = ["time"]
+    d2 = ["time", "level"]
+    d3 = ["time", "level", "level"]
+
+
+class Units(Enum):
+    time = "days since 1858-11-17 00:00"
+    altitude = "m"
+    lat = "degrees north"
+    lon = "degrees east"
+    hours = "hours"
+    unitless = "-"
+    pressure = "Pa"
+    temperature = "K"
+    degrees = "degrees"
+    koverk = "K/K"
+    poverp = "%/%"
+    product = "product"
+
+
+@attr.s
+class Filter:
+    residual = attr.ib(type=float)
+    minlmfactor = attr.ib(type=float)
+
+
+@attr.s
+class Parameter:
+    name = attr.ib(type=str)
+    units = attr.ib(type=Units)
```

Comment:
```suggestion
    unit = attr.ib(type=Units)
```

Reply:
Unless a `Parameter` can have multiple units.

Reply:
the netcdf keyword for units is units, thats why I used it. can change it but I do not comit here because I have to change tests.

Reply:
then it makes sense on some level even if it's a bit odd in this code.